### PR TITLE
Added spellCheckConfiguration to the constructor of CupertinoTextFormFieldRow

### LINKED ISSUE #159382: Adding spellCheckConfiguration to CupertinoTextFormFieldRow constructor
### Use case

Currently **CupertinoTextFormFieldRow** has no property to set the _spellCheckConfiguration_. This appears to be an oversight. Behind the scenes the CupertinoTextFormFieldRow creates a CupertinoTextField.borderless, which allows for that to be set, and defaults to SpellCheckConfiguration.disabled() if null, which will always be the case when using a CupertinoTextFormFieldRow.

Currently possible:

```dart
CupertinoTextField(
     spellCheckConfiguration: const SpellCheckConfiguration(),
 );
```

Not currently possible:

```dart
CupertinoTextFormFieldRow(
    spellCheckConfiguration: const SpellCheckConfiguration(),
);
```

### Proposal

Add the _spellCheckConfiguration_ property to the constructor for **CupertinoTextFormFieldRow**, and pass it through to the CupertinoTextField.borderless that is created.

### PR DESCRIPTION
**Pull request by sargntpi**
Before: No way to set the _spellCheckConfiguration_ property of a CupertinoTextFormFieldRow, meaning no spellcheck/highlighting of incorrect words is present. This can be done on the standard CupertinoTextField widget, but not the CupertinoTextFormFieldRow widget.
image

After: Adding this property allows the _spellCheckConfiguration_ to be configured and spellcheck/highlighting of incorrect words working as expected.
image
image

Fixes #159382 by adding the _ spellCheckConfiguration_ field to the constructor of CupertinoTextFormFieldRow and passing the value through to CupertinoTextField.borderless internally.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Comment by piedcipher** [comment]
Thanks for submitting the PR. As the bot says we'll need a test in order to land this.

**Comment by sargntpi** [comment]
> Thanks for submitting the PR. As the bot says we'll need a test in order to land this.

Thank you, and sorry, it's my first attempt at a PR. I've added a test to ensure it's still present, please let me know if I've missed or need to change anything.

**Review by piedcipher — APPROVED** [review]

**Review by Renzo-Olivares — APPROVED** [review]
LGTM, thank you for the contribution!

File: packages/flutter/lib/src/cupertino/text_form_field_row.dart [hunk 1]
```diff
@@ -234,6 +235,7 @@ class CupertinoTextFormFieldRow extends FormField<String> {
                   placeholder: placeholder,
                   placeholderStyle: placeholderStyle,
                   contextMenuBuilder: contextMenuBuilder,
+                  spellCheckConfiguration: spellCheckConfiguration
```
Comment (nate-thegrate) [inline packages/flutter/lib/src/cupertino/text_form_field_row.dart:238]: I _would_ do a trailing comma nitpick, but the autoformatter pass on the 19th will take care of it!

**Review by nate-thegrate — APPROVED** [review]
LGTM too!

**Pull request merged**